### PR TITLE
Add -DAPP=<app_name> cmake option. Add timing info to rt logs. 

### PR DESCRIPTION
**Pull request by DusanJovic-NOAA**
## Description

Add timing info for all compile jobs to the main log file.
Fix Rocoto option.
Save timestamps for each rt job

As the number of applications within the UFS grows, this PR collects the applications and re-defines it in terms of components that need to be enabled/disabled for the application.  It also allows the user the ability to define their own application based on a unique App Name.

The currently supported valid applications are:
|App Name| App Description| Components this App will turn ON|
|----------|-----------------|------------------|
| ATM    | Atmosphere Only | FMS, FV3 |
| ATMW | Atmosphere with Waves | FMS, FV3, WW3 |
| S2S    |  Coupled model | FMS, FV3, MOM6, CICE6, CMEPS |
| S2SW | Coupled model with Waves | FMS, FV3, MOM6, CICE6, WW3, CMEPS |
| DATM | Coupled Model with (CDEPS) Data atmosphere | FMS, MOM6, CICE6, CMEPS, CDEPS|
| DATM_NEMS | Coupled Model with (NEMS) Data atmosphere | FMS, MOM6, CICE6, CMEPS, NEMSdatm|

CMake expects an option `-DAPP=<APP_NAME>`, where `APP_NAME` is the item in the 1st column from the above table.
Without a valid application, the configuration will exit with a message.

To add new applications, the user will define a unique application name in  the list defined as `VALID_APPS` in `CMakeLists.txt` and proceed to enable the components for the new application in `cmake/configure_apps.cmake`.

## Testing

How were these changes tested? tested on hera.intel
What compilers / HPCs was it tested with? intel
Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)
Have regression tests and unit tests (utests) been run? yes
On which platforms and with which compilers? hera, intel

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [ ] wcoss_cray
- [ ] wcoss_dell_p3

Two tests failed on wcoss (fv3_gfs_v15p2 and fv3_gfs_v15p2_RRTMGP).

## Dependencies
No.

Do PRs in upstream repositories need to be merged first?
No.

### CODE REVIEW

**Comment by BrianCurtis-NOAA** [comment]
Machine: gaea
Compiler: intel
Repo location: /lustre/f2/pdata/ncep/Brian.Curtis/test/594080688/20210322111506/ufs-weather-model
Please manually delete: /lustre/f2/scratch/Brian.Curtis/FV3_RT/rt_31109
Test datm_debug_cfsr 102 failed in check_result failed 
Test datm_debug_cfsr 102 failed in run_test failed 
Please make changes and add the following label back:
gaea-intel-RT

**Comment by DusanJovic-NOAA** [comment]
> Machine: gaea
> Compiler: intel
> Repo location: /lustre/f2/pdata/ncep/Brian.Curtis/test/594080688/20210322111506/ufs-weather-model
> Please manually delete: /lustre/f2/scratch/Brian.Curtis/FV3_RT/rt_31109
> Test datm_debug_cfsr 102 failed in check_result failed
> Test datm_debug_cfsr 102 failed in run_test failed
> Please make changes and add the following label back:
> gaea-intel-RT

```
$ cat log_gaea.intel/rt_102_datm_debug_cfsr.log

baseline dir = /lustre/f2/pdata/esrl/gsd/ufs/ufs-weather-model/RT/NEMSfv3gfs/develop-20210318/INTEL/datm_debug_cfsr
working dir  = /lustre/f2/scratch/Brian.Curtis/FV3_RT/rt_31109/datm_debug_cfsr
Checking test 102 datm_debug_cfsr results ....
 Comparing RESTART/MOM.res.nc ............ALT CHECK......NOT OK
 Comparing RESTART/iced.2011-10-01-21600.nc ............ALT CHECK......NOT OK
 Comparing RESTART/DATM_CFSR.cpl.r.2011-10-01-21600.nc ............ALT CHECK......NOT OK

The total amount of wall time                        = 43.506641

Test 102 datm_debug_cfsr FAIL
```

**Comment by DusanJovic-NOAA** [comment]
The last `COMPILE` job in `rt.conf` is missing  `DEBUG=Y`

**Comment by BrianCurtis-NOAA** [comment]
Machine: orion
Compiler: intel
Repo location: /work/noaa/nems/emc.nemspara/autort/pr/594080688/20210322101513/ufs-weather-model
Please manually delete: /work/noaa/stmp/bcurtis/stmp/bcurtis/FV3_RT/rt_270862
Test fv3_gfdlmprad 017 failed in check_result failed 
Test fv3_gfdlmprad 017 failed in run_test failed 
Test fv3_gfdlmprad_atmwav 018 failed in check_result failed 
Test fv3_gfdlmprad_atmwav 018 failed in run_test failed 
Test fv3_wrtGauss_nemsio_c768 019 failed in check_result failed 
Test fv3_wrtGauss_nemsio_c768 019 failed in run_test failed 
Test datm_debug_cfsr 112 failed in check_result failed 
Test datm_debug_cfsr 112 failed in run_test failed 
Test cpld_control_wave 102 failed in check_result failed 
Test cpld_control_wave 102 failed in run_test failed 
Test cpld_bmarkfrac_wave_v16 101 failed in check_result failed 
Test cpld_bmarkfrac_wave_v16 101 failed in run_test failed 
Test cpld_bmark_wave 099 failed in check_result failed 
Test cpld_bmark_wave 099 failed in run_test failed 
Test cpld_bmarkfrac_wave 100 failed in check_result failed 
Test cpld_bmarkfrac_wave 100 failed in run_test failed 
Please make changes and add the following label back:
orion-intel-RT

**Comment by BrianCurtis-NOAA** [comment]
Machine: hera
Compiler: intel
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/594080688/20210322153015/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_15470
Test fv3_gfdlmprad 017 failed in check_result failed 
Test fv3_gfdlmprad 017 failed in run_test failed 
Test fv3_gfdlmprad_atmwav 018 failed in check_result failed 
Test fv3_gfdlmprad_atmwav 018 failed in run_test failed 
Test datm_debug_cfsr 112 failed in check_result failed 
Test datm_debug_cfsr 112 failed in run_test failed 
Test fv3_wrtGauss_nemsio_c768 019 failed in check_result failed 
Test fv3_wrtGauss_nemsio_c768 019 failed in run_test failed 
Please make changes and add the following label back:
hera-intel-RT

File: tests/compile.sh [hunk 1]
```diff
@@ -131,20 +131,30 @@ if [[ $? -eq 0 ]]; then
 fi
 set -ex
 
-if [[ "${MAKE_OPT}" == *"WW3=Y"* ]]; then
-    CMAKE_FLAGS="${CMAKE_FLAGS} -DWW3=Y"
+# Valid applications
```
Comment (JessicaMeixner-NOAA) [inline tests/compile.sh:134]: Did this option get added somewhere else? Or did how WW3 get added to the CMAKE_FLAGS change?

**Comment by junwang-noaa** [comment]
WW3 is set up at line 9 in cmake/configure_apps.cmake
<https://github.com/ufs-community/ufs-weather-model/pull/477/files#diff-54eb4291b1e64abd286066214745be361a03108a0eea03e8d8f44504dfb258d7>

**Comment by BrianCurtis-NOAA** [comment]
Machine: orion
Compiler: intel
Repo location: /work/noaa/nems/emc.nemspara/autort/pr/594080688/20210322183014/ufs-weather-model
Please manually delete: /work/noaa/stmp/bcurtis/stmp/bcurtis/FV3_RT/rt_274257
Test fv3_gfsv16_csawmg 052 failed failed 
Test fv3_gfsv16_csawmg 052 failed in run_test failed 
Test cpld_bmark_wave 099 failed in run_test failed 
Test cpld_bmarkfrac_wave 100 failed in run_test failed 
Please make changes and add the following label back:
orion-intel-RT

**Review by climbfuji — APPROVED** [review]
Is there a default for `APP` or should there be one? (`APP=ATM`).

This PR does a ton more than what is in the title or the description. It would be great to add a bit of information so that we can find those PRs easier if needed in the future.

**Comment by MinsukJi-NOAA** [comment]
CI errors have to do with EC2 instance, and not ufs-weather-model. I am looking into this.

**Comment by MinsukJi-NOAA** [comment]
> CI errors have to do with EC2 instance, and not ufs-weather-model. I am looking into this.

Should be good to go now.

**Review by junwang-noaa — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -36,6 +36,10 @@ E.g. use of `build.sh` to build the coupled model with `FV3_GFS_v15p2` as the CC
 ```
 $> CMAKE_FLAGS="-DS2S=ON" CCPP_SUITES="FV3_GFS_v15p2" ./build.sh
```
Comment (aerorahul) [inline README.md]: I think we need to update the `README.md` to replace `-DS2S=ON` with `-DAPP=S2S`

File: tests/compile.sh (hunk 1 above)
Comment (aerorahul) [reply tests/compile.sh:134]: @JessicaMeixner-NOAA 
`-DWW3=ON|OFF` is no longer an option a user should set.
They should define an application `S2SW` and `ATMW` will turn ON the application with waves.
The corresponding applications without WW3 are `S2S` and `ATM` without the `W`.

File: README.md (hunk 2 above)
Comment (aerorahul) [inline README.md]: Please make this change in `README.md`

**Review by aerorahul — CHANGES_REQUESTED** [review]
All changes look good. 
One change in `README.md` is requested.

**Review by aerorahul — APPROVED** [review]
Looks good.
Thanks for incorporating changes from PR #473. 
PR #473 address and closes #416

**Pull request merged**